### PR TITLE
Fix "field" action - return all records, allow Action as scope condition

### PR DESCRIPTION
**Pull request by mvorisek**
merge after https://github.com/atk4/data/pull/860

also fixes array persistence `tryLoad` which previously was completely ignoring all conditions (they were used only when loading data using iterator)

### CODE REVIEW

**Comment by mvorisek** [comment]
@georgehristov can you please review the last commit?

**Comment by mvorisek** [comment]
@georgehristov can you please review?

Do you know why https://github.com/atk4/data/pull/861/commits/7d598f19ce77a9356a1388e5e32a86002aa603b3#diff-facfb8206c512f6ada3a34d73c4ed464957c84987a77f986e2b8357bca34ab29R205 is needed to NOT check keys? The scope is for `refId` but the data have `id` keys. Are (our/their) column names/aliases set correctly in HasMany class?

File: src/Persistence/Array_.php [hunk 1]
```diff
@@ -185,38 +197,35 @@ public function add(Model $model, array $defaults = []): Model
...
-        if (!isset($this->data[$model->table])) {
-            throw (new Exception('Table was not found in the array data source'))
-                ->addMoreInfo('table', $model->table);
-        }
+        $table = $this->seedDataAndGetTable($model);
 
         if ($id === self::ID_LOAD_ONE || $id === self::ID_LOAD_ANY) {
-            if (count($this->data[$model->table]) === 0) {
+            $action = $this->action($model, 'select');
+            $action->generator->rewind(); // needed for some reasons!
```
Comment (mvorisek) [inline src/Persistence/Array_.php]: why is this needed, should this be done in the action itself?

**Review by DarkSide666 — APPROVED** [review]
LGTM, but would be good if @georgehristov could take a look too
